### PR TITLE
Added additional OIDs for CMS signatures.

### LINKED ISSUE #32639: SignedCms compatibility for the SignerInfo SignatureAlgorithm issue
I've observed another divergence of behavior from `SignedCms` between netcoreapp2.1 and net472.

If the CMS has a SignerInfo with a `SignatureAlgorithm` OID of "1.2.840.113549.1.1.11" (sha256WithRSAEncryption), an exception is raised:

>CryptographicException: 'Unknown algorithm '1.2.840.113549.1.1.11'.'

It appears that `CmsSignature` is only expecting '1.2.840.113549.1.1.1' which is just RSA Encryption:

https://github.com/dotnet/corefx/blob/0ce22c97d7117d2cf974e1da40bb4ba72fa53e65/src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.RSA.cs#L18-L19

This behavior diverges with the Desktop .NET Framework (CAPI) which is tolerant of SignerInfos with a signature algorithm OID 1.2.840.113549.1.1.11 (and presumably 1.2.840.113549.1.1.12, etc for SHA384).

I've put together a small repro issue over at https://github.com/vcsjones/SignedCmsBug.

This succeeds:

```
dotnet run --framework net472
```

this fails:

```
dotnet run --framework netcoreapp2.1
```

This is using .NET Core 2.1.5.

/cc @bartonjs

### PR DESCRIPTION
**Pull request by vcsjones**
This adds additional OIDs for RSA CMS validation when the OID is specific about which hash algorithm should be used.

This is a slightly different approach from the DSA / ECDsa approach. This allows for a null `HashAlgorithmName` and does not enforce the hash algorithm if one is not specified when using the "1.2.840.113549.1.1.1" OID. 

Fixes #32639

### CODE REVIEW

**Comment by vcsjones** [comment]
Hm. Now that I think about it, we can do 1.2.840.113549.1.1.4 for MD5 support. @bartonjs, do we want an MD5WithRsa OID here?

**Comment by vcsjones** [comment]
Interesting failure for UWP:

>'SignerInfo' does not contain a definition for 'SignatureAlgorithm' and no accessible extension method 'SignatureAlgorithm' accepting a first argument of type 'SignerInfo' could be found (are you missing a using directive or an assembly reference?

Guess I can just remove the assertion. It was there as a sanity check.

File: src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs [hunk 1]
```diff
@@ -767,5 +767,123 @@ internal static class SignedDocuments
...
+            "980C5A30EC00729D1CFA826D7A65B43FF6806B5E0ABA23A78E4F1CAA3F6436EF" +
+            "00941C6947A9B8F20D0757B5346CF640AA217F7361BEEFF2BC997FB1D3597BF3" +
+            "D7457BD4A94062FB03660F9D86710BE2FC99876A848251F4965E1B16192714C8" +
+            "F9788C09CCDE83603ADC919297BA496E921B95F3BD9554A873E09912640FCFAA" +
+            "D9DD1441D1851E637031D390C038223AE64B048E806462DDBAC98C156BE2EE47" +
+            "2B78166BDB1612848B535ADC3F0E7BE52991A17F48AFDCCC1698A236BA338930" +
+            "50EBAAC4460DAA35185C16670F597E0E6E0CB0AA83F51AAEF452F3367DD9350A" +
+            "8A49A5A8F79DF8E921303AB5D6646A482F0F59D9980310E1AE3EE8D77CB857").HexToByteArray();
+
+        internal static readonly byte[] RsaPkcs1signedSha1DeclaredSha256WithRsa = (
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs]: the `S` in `signed` should be uppercase :)

**Comment by bartonjs** [comment]
> @bartonjs, do we want an MD5WithRsa OID here?

Eh, I'd leave it out unless there's a reason to bother later.

File: src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs (hunk 1 above)
Comment (vcsjones) [reply src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedDocuments.cs]: Doh. 🐪

**Comment by vcsjones** [comment]
>Eh, I'd leave it out unless there's a reason to bother later.

Makes sense. I have no compelling reason to put it there other than completeness, and I'm not particularly fond of MD5-anything.

Fixed the casing issue.

**Comment by bartonjs** [comment]
@dotnet-bot Test this please (lots of failures due to what looks like a Jenkins restart)

**Comment by bartonjs** [comment]
@vcsjones  The (valid) test document you added doesn't have indefinite length content, right?  (I don't see an 0x24, so it shouldn't be; but making sure I'm not being redundant)

**Comment by vcsjones** [comment]
@bartonjs no they do not, I didn't want to risk cramming two issues into the same CMS.

**Pull request merged**